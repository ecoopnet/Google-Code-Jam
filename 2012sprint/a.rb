#!/usr/bin/env ruby -Ku
# Usage: cat problems | program.rb

class Solver
	def solve line
		items = line.split(" ")
		@L = items.shift.to_i
		@M = items.shift.to_i
		@exp = items.sort.reverse
		@exp.delete("0")
		if @exp.length == 0
			return 0
		end
		@exp.each do |i|
			if i.to_i > @L
				return -1;
			end
		end
		min = @exp.pop.to_i
		if @L == 0
			return 0
		end

		#print @exp,min,"/",@M,"/",@L,"\n"
		res = 0
		if @M > 1
			res+=(@M-1)*@L
#			@exp.each do |i|
#				res+= (i.to_i * @L)
#			end
		end
		#print min,"\n"
		if (min > 0)
			res+=min
		end
		return res;
	end
end

solver = Solver.new
number = ARGF.gets
i = 0
while line = ARGF.gets
	line.strip!
	if line == ""
		next
	end
	i+=1
	print "Case #",i,": ", solver.solve(line),"\n"
end
