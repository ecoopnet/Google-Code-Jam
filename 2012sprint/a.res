Case #1: 12
Case #2: 3
Case #3: 2
Case #4: 1
Case #5: 9
Case #6: 0
Case #7: 2
Case #8: -1
Case #9: 11
Case #10: 0
Case #11: 0
Case #12: 5
Case #13: -1
Case #14: 9
Case #15: 3
Case #16: 2
Case #17: 5
Case #18: 12
Case #19: 3
Case #20: -1
Case #21: 2
Case #22: 0
Case #23: -1
Case #24: 11
Case #25: 1
Case #26: 1
Case #27: 1
Case #28: -1
Case #29: 1
Case #30: 20
Case #31: 0
Case #32: 11
Case #33: 13
Case #34: 1
Case #35: 2
Case #36: 1
Case #37: 0
Case #38: 0
Case #39: 3
Case #40: 0
Case #41: -1
Case #42: 0
Case #43: 13
Case #44: 0
Case #45: 11
Case #46: 1
Case #47: 3
Case #48: 12
Case #49: 3
Case #50: 5
Case #51: 0
Case #52: 1
Case #53: 0
Case #54: 0
Case #55: 0
Case #56: 0
Case #57: 1
Case #58: 9
Case #59: 3
Case #60: 8
Case #61: 1
Case #62: 3
Case #63: 0
Case #64: 0
Case #65: 0
Case #66: 10
Case #67: 8
Case #68: 0
Case #69: 10
Case #70: 1
Case #71: -1
Case #72: 1
Case #73: -1
Case #74: -1
Case #75: 2
Case #76: 0
Case #77: 0
Case #78: 10
Case #79: 0
Case #80: 0
Case #81: 12
Case #82: 0
Case #83: 5
Case #84: 2
Case #85: 3
Case #86: -1
Case #87: 2
Case #88: 0
Case #89: 1
Case #90: 7
Case #91: 1
Case #92: 3
Case #93: 0
Case #94: 0
Case #95: 0
Case #96: -1
Case #97: 0
Case #98: 1
Case #99: 0
Case #100: 2
