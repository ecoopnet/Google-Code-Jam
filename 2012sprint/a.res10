Case #1: 1
Case #2: 0
Case #3: 4
Case #4: 0
Case #5: 2
Case #6: 0
Case #7: 1
Case #8: 5
Case #9: 0
Case #10: 1
Case #11: 0
Case #12: 3
Case #13: -1
Case #14: 0
Case #15: 8
Case #16: 3
Case #17: 0
Case #18: 0
Case #19: 4
Case #20: 0
Case #21: 7
Case #22: 5
Case #23: 2
Case #24: 0
Case #25: 4
Case #26: -1
Case #27: 0
Case #28: 6
Case #29: 2
Case #30: 0
Case #31: -1
Case #32: 4
Case #33: -1
Case #34: -1
Case #35: 3
Case #36: 0
Case #37: 0
Case #38: 1
Case #39: 0
Case #40: 0
Case #41: 0
Case #42: 1
Case #43: -1
Case #44: 6
Case #45: 4
Case #46: 9
Case #47: 6
Case #48: 0
Case #49: 3
Case #50: 0
Case #51: 0
Case #52: 2
Case #53: -1
Case #54: 2
Case #55: 5
Case #56: 0
Case #57: 6
Case #58: 8
Case #59: 5
Case #60: 4
Case #61: 0
Case #62: 9
Case #63: 2
Case #64: 3
Case #65: 2
Case #66: 2
Case #67: 3
Case #68: 4
Case #69: 1
Case #70: 2
Case #71: 0
Case #72: -1
Case #73: 0
Case #74: -1
Case #75: -1
Case #76: 6
Case #77: -1
Case #78: -1
Case #79: 9
Case #80: 3
Case #81: 3
Case #82: 0
Case #83: 7
Case #84: 0
Case #85: 6
Case #86: 0
Case #87: -1
Case #88: 0
Case #89: 2
Case #90: 9
Case #91: 0
Case #92: 0
Case #93: -1
Case #94: 9
Case #95: 6
Case #96: 4
Case #97: 5
Case #98: -1
Case #99: 5
Case #100: 0
