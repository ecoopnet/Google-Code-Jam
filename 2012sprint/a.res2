Case #1: 5
Case #2: 4
Case #3: -1
Case #4: 9
Case #5: 0
Case #6: 7
Case #7: 0
Case #8: 0
Case #9: 1
Case #10: 0
Case #11: -1
Case #12: 0
Case #13: 1
Case #14: 6
Case #15: 0
Case #16: -1
Case #17: 1
Case #18: 0
Case #19: 0
Case #20: 0
Case #21: 0
Case #22: 0
Case #23: 0
Case #24: 0
Case #25: 6
Case #26: 8
Case #27: 5
Case #28: 3
Case #29: 2
Case #30: 1
Case #31: 0
Case #32: 2
Case #33: 7
Case #34: 0
Case #35: 3
Case #36: 2
Case #37: 8
Case #38: 0
Case #39: 3
Case #40: 0
Case #41: 6
Case #42: 1
Case #43: 7
Case #44: 0
Case #45: 5
Case #46: 0
Case #47: 0
Case #48: 7
Case #49: 0
Case #50: 1
Case #51: 9
Case #52: 1
Case #53: 0
Case #54: 9
Case #55: 0
Case #56: 6
Case #57: 3
Case #58: 4
Case #59: 2
Case #60: 5
Case #61: 2
Case #62: 4
Case #63: 2
Case #64: 0
Case #65: 2
Case #66: 7
Case #67: -1
Case #68: 0
Case #69: 7
Case #70: 6
Case #71: 0
Case #72: 0
Case #73: 0
Case #74: 0
Case #75: 0
Case #76: 0
Case #77: 0
Case #78: 3
Case #79: -1
Case #80: 3
Case #81: 0
Case #82: 5
Case #83: 0
Case #84: 1
Case #85: 2
Case #86: 3
Case #87: 0
Case #88: 5
Case #89: 8
Case #90: 3
Case #91: 3
Case #92: -1
Case #93: 8
Case #94: 2
Case #95: 5
Case #96: 0
Case #97: 2
Case #98: 0
Case #99: 0
Case #100: 2
