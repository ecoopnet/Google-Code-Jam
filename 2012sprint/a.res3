Case #1: 6
Case #2: 0
Case #3: 6
Case #4: 3
Case #5: 7
Case #6: 8
Case #7: 0
Case #8: 0
Case #9: -1
Case #10: 0
Case #11: 1
Case #12: 0
Case #13: 0
Case #14: 0
Case #15: 0
Case #16: 5
Case #17: 3
Case #18: 4
Case #19: 0
Case #20: 0
Case #21: 5
Case #22: 8
Case #23: 0
Case #24: 5
Case #25: 4
Case #26: -1
Case #27: 0
Case #28: 4
Case #29: 2
Case #30: 2
Case #31: 0
Case #32: 2
Case #33: 0
Case #34: 2
Case #35: 1
Case #36: 5
Case #37: 0
Case #38: 0
Case #39: 0
Case #40: 1
Case #41: 0
Case #42: 7
Case #43: 3
Case #44: 7
Case #45: 9
Case #46: 2
Case #47: -1
Case #48: 0
Case #49: 2
Case #50: 0
Case #51: 7
Case #52: -1
Case #53: 0
Case #54: 0
Case #55: 0
Case #56: -1
Case #57: 6
Case #58: 0
Case #59: 0
Case #60: 1
Case #61: 6
Case #62: 3
Case #63: 1
Case #64: 2
Case #65: 0
Case #66: 2
Case #67: 1
Case #68: 0
Case #69: 3
Case #70: 0
Case #71: 0
Case #72: 5
Case #73: 7
Case #74: 5
Case #75: 0
Case #76: 5
Case #77: 1
Case #78: 3
Case #79: 0
Case #80: 1
Case #81: 8
Case #82: 2
Case #83: 1
Case #84: 0
Case #85: 4
Case #86: 0
Case #87: 4
Case #88: 6
Case #89: 2
Case #90: 0
Case #91: 8
Case #92: 0
Case #93: 2
Case #94: 7
Case #95: 0
Case #96: 2
Case #97: 3
Case #98: 0
Case #99: 0
Case #100: -1
