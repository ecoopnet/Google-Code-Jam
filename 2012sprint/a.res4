Case #1: 6
Case #2: 5
Case #3: -1
Case #4: 5
Case #5: 3
Case #6: 3
Case #7: -1
Case #8: 2
Case #9: 5
Case #10: 1
Case #11: 3
Case #12: 6
Case #13: 2
Case #14: 5
Case #15: 6
Case #16: 1
Case #17: 5
Case #18: 4
Case #19: 4
Case #20: 1
Case #21: -1
Case #22: 3
Case #23: 7
Case #24: 3
Case #25: -1
Case #26: 6
Case #27: -1
Case #28: -1
Case #29: 5
Case #30: 3
Case #31: 8
Case #32: -1
Case #33: -1
Case #34: -1
Case #35: -1
Case #36: 7
Case #37: -1
Case #38: -1
Case #39: 2
Case #40: 7
Case #41: -1
Case #42: 9
Case #43: 5
Case #44: -1
Case #45: 2
Case #46: -1
Case #47: 8
Case #48: 1
Case #49: 3
Case #50: 3
Case #51: -1
Case #52: -1
Case #53: 3
Case #54: 1
Case #55: -1
Case #56: 3
Case #57: 1
Case #58: 4
Case #59: -1
Case #60: 4
Case #61: 1
Case #62: -1
Case #63: 3
Case #64: -1
Case #65: -1
Case #66: -1
Case #67: 9
Case #68: -1
Case #69: -1
Case #70: -1
Case #71: 1
Case #72: 2
Case #73: 1
Case #74: 3
Case #75: -1
Case #76: 5
Case #77: 1
Case #78: 5
Case #79: 2
Case #80: -1
Case #81: 2
Case #82: 5
Case #83: -1
Case #84: 3
Case #85: -1
Case #86: 8
Case #87: 7
Case #88: 2
Case #89: -1
Case #90: 3
Case #91: 9
Case #92: -1
Case #93: -1
Case #94: 5
Case #95: 5
Case #96: -1
Case #97: -1
Case #98: -1
Case #99: 6
Case #100: -1
