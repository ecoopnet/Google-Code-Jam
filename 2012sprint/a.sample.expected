ase #1: 5
Case #2: -1
